src/libraries/System.Net.NameResolution/src/System/Net/NameResolutionPal.Windows.cs

```diff
@@ -194,10 +194,10 @@ private static unsafe void GetAddressInfoExCallback(int error, int bytes, Native
 
         private static unsafe void ProcessResult(SocketError errorCode, GetAddrInfoExContext* context)
         {
+            GetAddrInfoExState state = GetAddrInfoExState.FromHandleAndFree(context->QueryStateHandle);
```

Comment:
Why it is not inside `try` anymore?

Reply:
Because `state` is being disposed now in the finally block, so it has to live out the `try` scope. `FromHandleAndFree` should only throw if there is a bug in our code.